[JENKINS-54972] Include the SCM name on changes pages

richardwhiuk opened the pull request:
## [JENKINS-54972](https://issues.jenkins-ci.org/browse/JENKINS-54972) - Include SCM Name on changes pages

There's a bunch of places where it'd be useful to expose the custom SCM name if one is configured, especially when working with multi-repo build jobs.

This exposes it on `/job/<build>/changes` and on `/job/<build>/<build_number>/changes`.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. This is simply a reminder of what we are going to look for before merging your code._

- [X] I have read the [CONTRIBUTING](https://github.com/jenkinsci/git-plugin/blob/master/CONTRIBUTING.md) doc
- [X] I have referenced the Jira issue related to my changes in one or more commit messages
- [ ] I have added tests that verify my changes
- [X] Unit tests pass locally with my changes
- [X] I have added documentation as necessary
- [X] No Javadoc warnings were introduced with my changes
- [X] No findbugs warnings were introduced with my changes
- [X] I have interactively tested my changes
- [X] Any dependent changes have been merged and published in upstream modules (like git-client-plugin)

## Types of changes

What types of changes does your code introduce? _Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Further comments

The SCM name tests in `hudson.plugins.git.GitSCMTest.testCustomSCMName` appear to be commented out - I think that's the logical place to add a test if required (where the test would probably just do something along the lines of `expectedScmName == createChangeLogParser().parse().getScmName()` ).

I've tested it by installing a plugin with the changes on one of Jenkins servers.

MarkEWaite commented:
Sorry for the conflicting change in the GitChangeLogParser for JENKINS-29977.  Is the scmName already included in the data tracked by GitClient?  If so, could you adapt to use the new GitChangeLogParser(GitClient, boolean) and then extract the scmName from the passed git client?

richardwhiuk commented:
@MarkEWaite Thanks for the feedback - sorry for the delay on getting back to you on this.

I've rebased the branch off master, and fixed the merge conflicts.

Unfortunately no, the SCM Name isn't part of the Git Client - it's only known by the GitSCM object - and short of passing that into the change log parser, I think this is the sensible fix.

Can you let me know if there's anything else blocking this being merged?
